src/dotnet/commands/dotnet-projectmodel-server/ProjectManager.cs

```diff
@@ -200,8 +200,10 @@ private bool ProcessMessage()
                     _configure.Value = message.Payload.GetValue("Configuration");
                     break;
                 case MessageTypes.RefreshDependencies:
+                    _refreshDependencies.Value = false;
+                    break;
                 case MessageTypes.RestoreComplete:
-                    _refreshDependencies.Value = 0;
+                    _refreshDependencies.Value = message.Payload.HasValues ? message.Payload.Value<bool>("Reset") : true;
```

Comment:
This is a bit off convention. Booleans should default to false in .NET. It would be nice to get this to match that expectation if possible.